addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -35,6 +35,7 @@ const preserveCursor = OdooEditorLib.preserveCursor;
 const closestElement = OdooEditorLib.closestElement;
 const setSelection = OdooEditorLib.setSelection;
 const endPos = OdooEditorLib.endPos;
+const getCursorDirection = OdooEditorLib.getCursorDirection;
```

Comment:
Commit message:

```diff
@@ -2,14 +2,24 @@ Before this commit:
 
 `destroyLinkTools` function sets the selection to entire link. However, in
 case where a website snippet had a structure like
-`<a><div><i class=fa-xxx>​</i><div><h4>Text</h4><font>Text</font></div></div>
-</a>` selecting the complete link caused problem. The toolbar couldn't be
+```html
+<a>
+    <div>
+        <i class=fa-xxx>\u200b​</i>
+        <div>
+            <h4>Text</h4>
+            <font>Text</font>
+        </div>
+    </div>
+</a>
+```
+selecting the complete link caused problem. The toolbar couldn't be
 updated correctly, also one could not change the a tag of a single element
 within the link.
 
 After this commit:
 
-Fix destroyLinkTools such that now it select the anchorNode and the focus
-node of selection instead of complete link.
+`destroyLinkTools` selects the `anchorNode` and the `focusNode` of the
+selection instead of the entire link.
 
 task-3245819
```